examples/llama/train_llama2_fsdpv2.sh

```diff
@@ -0,0 +1,342 @@
+
+# parsing input arguments
+for ARGUMENT in "$@"
+do
+   KEY=$(echo $ARGUMENT | cut -f1 -d=)
+
+   KEY_LENGTH=${#KEY}
+   VALUE="${ARGUMENT:$KEY_LENGTH+1}"
+
+   export "$KEY"="$VALUE"
+done
+
+
+TIME_STAMP=$(date +"%Y-%m-%d_%H-%M-%S")
+EXP_NAME="${EXP_NAME:-perf}"
+
+NCCL_SOCKET_IFNAME="${NCCL_SOCKET_IFNAME:-ens51np0}"
+GLOO_SOCKET_IFNAME="${GLOO_SOCKET_IFNAME:-ens51np0}"
+TEE_OUTPUT="${TEE_OUTPUT:-1}"
+NO_TORCH_COMPILE="${NO_TORCH_COMPILE:-1}"
+USE_FLASH_ATTN="${USE_FLASH_ATTN:-1}"
+NO_TRAINING="${NO_TRAINING:-0}" # NO_TRAINING=1: for computing metrics only
+ENABLE_PROFILING="${ENABLE_PROFILING:-0}"
+echo "NO_TRAINING=$NO_TRAINING"
+
+CWD=`pwd`
+GPUS_PER_NODE=`python3 -c "import torch; print(torch.cuda.device_count())"`
+# Change for multinode config
+MASTER_ADDR="${MASTER_ADDR:-localhost}"
+MASTER_PORT="${MASTER_PORT:-23731}"
+NNODES="${NNODES:-1}"
+NODE_RANK="${NODE_RANK:-0}"
+WORLD_SIZE=$(($GPUS_PER_NODE*$NNODES))
+
+MODEL_SIZE="${MODEL_SIZE:-70}"
+TP="${TP:-1}"
+PP="${PP:-1}"
+CP="${CP:-1}"
+MBS="${MBS:-1}"
+BS="${BS:-8}"
+SEQ_LENGTH="${SEQ_LENGTH:-2048}"
+TOTAL_ITERS="${TOTAL_ITERS:-20}"
+CONTI_PARAMS="${CONTI_PARAMS:-0}"
+OPTIMIZER="${OPTIMIZER:-adam}"
+TE_FP8="${TE_FP8:-0}"
+GEMM_TUNING="${GEMM_TUNING:-1}"
+MCORE="${MCORE:-1}"
+FSDP="${FSDP:-1}"
+
+EXPERIMENT_DIR="experiment"
+mkdir -p $EXPERIMENT_DIR
+
+DATA_DIR="${DATA_DIR:-/root/.cache/data}"
+DATA_PATH="${DATA_PATH:-"$DATA_DIR/wikipedia_20220301.en/wikipedia_20220301.en.train.jsonl_text_document"}"
+
+TOKENIZER_MODEL="${TOKENIZER_MODEL:-"$DATA_DIR/tokenizer_llama2"}"
+# Download the tokenizer model
+if ! [ -d "$TOKENIZER_MODEL" ]; then
+  mkdir -p $TOKENIZER_MODEL
+  wget -O $TOKENIZER_MODEL/special_tokens_map.json https://huggingface.co/NousResearch/Llama-2-7b-chat-hf/resolve/main/special_tokens_map.json
+  wget -O $TOKENIZER_MODEL/tokenizer.json https://huggingface.co/NousResearch/Llama-2-7b-chat-hf/resolve/main/tokenizer.json
+  wget -O $TOKENIZER_MODEL/tokenizer.model https://huggingface.co/NousResearch/Llama-2-7b-chat-hf/resolve/main/tokenizer.model
+  wget -O $TOKENIZER_MODEL/tokenizer_config.json https://huggingface.co/NousResearch/Llama-2-7b-chat-hf/resolve/main/tokenizer_config.json
+
+  echo "Tokenizer files downloaded successfully to $TOKENIZER_MODEL."
+else
+  echo "Folder $TOKENIZER_MODEL already exists. Skipping download."
+fi
+
+MAX_POSITION_EMBEDDINGS=128000
+
+DEFAULT_LOG_DIR="${EXPERIMENT_DIR}/${NNODES}nodes_rank${NODE_RANK}_train_${MODEL_SIZE}B_mbs${MBS}_bs${BS}_tp${TP}_pp${PP}_cp${CP}_iter${TOTAL_ITERS}/nocompile${NO_TORCH_COMPILE}_TE_FP8_${TE_FP8}/${TIME_STAMP}"
+LOG_DIR="${LOG_DIR:-${DEFAULT_LOG_DIR}}"
+TRAIN_LOG="${LOG_DIR}/output_${EXP_NAME}.log"
+mkdir -p $LOG_DIR
+echo $TRAIN_LOG
+
+if [ "$GEMM_TUNING" -eq 1 ]; then
+    export TE_HIPBLASLT_TUNING_RUN_COUNT=20
+    export TE_HIPBLASLT_TUNING_ALGO_COUNT=100
+fi
+
+if [ "$SEQ_LENGTH" -le 8192 ]; then
+  ds_works=8
+else
+  ds_works=24
+fi
+
+if [[ $MODEL_SIZE -eq 8 ]]; then #llama3.1-8B
```

Comment:
Looks like this is LLama 3.1 configuration. Can you please confirm and update your script name accordingly.